updates/v2.1.0/migrate_message_code.php

```diff
@@ -11,9 +12,29 @@ class MigrateMessageCode extends Migration
 
     public function up()
     {
+        
+        if (!Schema::hasColumn(self::TABLE_NAME, 'code_pre_2_1_0')) {
+            Schema::table(self::TABLE_NAME, function($table) {
+                $table->string('code_pre_2_1_0')->index()->nullable();
+            });
+        }
+        
         foreach (Message::all() as $message) {
-            $default_message = $message->message_data['x'];
-            $message->code = Message::makeMessageCode($default_message);
+            $message->code_pre_2_1_0 = $message->code;
+            $message->code = $message->code_md5 ?: Message::makeMessageCode($message->message_data[Message::DEFAULT_LOCALE]);
+            $message->save();
+        }
+        
+        if (Schema::hasColumn(self::TABLE_NAME, 'code_md5')) {
```

Comment:
Where does `code_md5` column comes from?

Reply:
it is created by `down()` to preserve md5 hashes on downgrade  (can be useful for manual review).

Reply:
Oh, I see. So why not use it if it exists instead of re-generate it, fair enough.

Reply:
That line is triggered only if `up()` -> `down()` -> `up()`, which is a marginal situation, but I would like to manage it.
I prefer to drop it on `up()` to keep table "clean" and avoid misleading in future.